lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -745,6 +749,33 @@ ChangeResult TypeAnalyzer::visitReductionAlongDimIntListOp(
   }
   return getLatticeElement(op->getResult(0)).join(knowledge);
 }
+ChangeResult TypeAnalyzer::visitAtenArgmaxOp(
+    AtenArgmaxOp op, ArrayRef<LatticeElement<ValueKnowledge> *> operands) {
+  auto input = operands[0]->getValue();
+  auto knowledge = ValueKnowledge::getPessimisticValueState(op->getContext());
+  knowledge.dtype = IntegerType::get(op->getContext(), 64, IntegerType::Signed);
```

Comment:
`si64` is correct here :+1:

Reply:
Seems the dtype can be other types as well? Should we just use the dtype from the input?

Reply:
argmax returns the indices of the value, not the value itself:
https://pytorch.org/docs/stable/generated/torch.argmax.html

Reply:
ha, I see.